text/0000-nested-publish.md

```diff
@@ -0,0 +1,359 @@
+
+    ```toml
+    [package]
+    # This package may be nested but may not be `cargo publish`ed by itself
+    publish.nested = true
+    ```
+
+
+    To allow both regular publishing and nested publishing, the currently-allowed values for `publish` (`publish = true` or `publish = [registries...]`) may be specified under the key `package.publish.registries`. Examples:.
+
+    ```toml
+    [package]
+    publish.registries = ["crates.io"]
+    publish.nested = true
+    ```
+
+    ```toml
+    [package]
+    publish.registries = true
+    publish.nested = true
+    ```
+
+    If `package.publish` is a table, then `package.publish.registries` defaults to `false`, regardless of the value or presence of `package.publish.nested`.
+
+    Note: This dual-publishing-mode functionality is permitted mainly to keep the functionality composable/orthogonal. We hope that in most cases, packages are either published nested exactly once, or to a registry alone, to avoid duplicating code in the registry and compiling it redundantly.
+
+* The `dependencies.*.publish` field is newly defined, with the only currently allowed value being `"nested"`, to declare that that dependency is a nested dependency.
+    * It is an error if a nested dependency does not have a `path` field, or if it has a `version`, `git`, or any other package source field, unless future work defines a meaning for that combination.
+    * Workspace inheritance is not permitted; the presence of `workspace.dependencies.*.publish` is an error.
+
+When a nested dependency is present (making its referent be a nested package), the following additional requirements apply:
+
+* The nested package must have `package.publish.nested = true`.
+* If the nested package specifies `package.license`, its value must be identical to the parent package's.
+
+  This check is intended only to prevent accidents (such as vendoring a third-party package without considering the implications of redistributing it). It is always valid to omit `package.license` from the nested package, thus making no machine-readable claims about its licensing.
+
+It is an error for a nested package to have the same package name as the parent package or any other nested package with the same parent package. This is validated by all Cargo operations that would generate or read a lockfile. Rationale: This should ensure that whenever a nested package must be named, such as in an `.crate` archive, potentially in lock files, and potentially in Cargo user interface, the pair of (parent package name, nested package name) is sufficient to uniquely identify the package.
```

Comment:
Should we just leave this as a conflict validation check during publish?  That means one less thing to have to validate on every command and this would allow nested packages in nested directories to not be subject to this limitation